src/controller/python/PyDelegateWrappers.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+#include "pybind11/pybind11.h"
+#include <controller/DeviceAddressUpdateDelegate.h>
+#include <controller/CHIPDeviceController.h>
+
+class PyDeviceAddressUpdateDelegate : public chip::Controller::DeviceAddressUpdateDelegate {
+public:
+    /* Inherit the constructors */
+    using chip::Controller::DeviceAddressUpdateDelegate::DeviceAddressUpdateDelegate;
+
+    /* Trampoline (need one for each virtual function) */
+    void OnAddressUpdateComplete(chip::NodeId nodeId, CHIP_ERROR erros) override {
```

Comment:
I think that GIL should be used here to protect python globals. because this function may not be called from python thread.

Reply:
I believe this is already handled in the following macro: https://pybind11.readthedocs.io/en/stable/reference.html#c.PYBIND11_OVERRIDE_PURE

Reply:
@krypton36 

https://pybind11.readthedocs.io/en/stable/advanced/misc.html#global-interpreter-lock-gil

check here, it is required to lock gil if the function is called outside python thread.

Reply:
Since the method signatures of Python and C++ are the same, the GIL lock is handled in the Macro above. If you override methods with different signatures, then `py::gil_scoped_acquire acquire;` is required. The macro handles this for these delegates because they are a 1:1 mapping of the virtual functions.

Reply:
After checking t he source code of pybind11, I found that gil is called in the macro, Great.